src/fsharp/FSharp.Core/option.fs

```diff
@@ -16,6 +16,18 @@ namespace Microsoft.FSharp.Core
         [<CompiledName("IsNone")>]
         let inline isNone option = match option with None -> true | Some _ -> false
 
+        [<CompiledName("DefaultIfNone")>]
+        let defaultIfNone def option = match option with None -> def | Some v -> v
```

Comment:
For use cases like this, is it more appropriate to give the JIT a hint to do the inlining with `AggressiveInlining`? I see the case for `inline` on `contains`, but still a bit unsure for `defaultValue` and `orElse`.